posts/2021/09/re-engineering-cicd-pipelines-for-scipy.md

```diff
@@ -0,0 +1,209 @@
+<!--
+.. title: Re-Engineering CI/CD pipelines for SciPy
+.. slug: re-engineering-cicd-pipelines-for-scipy
+.. date: 2021-09-29 05:51:58 UTC-05:00
+.. author: Harsh Mishra
+.. tags: SciPy, cicd, github-actions, internship-2021
+.. category: 
+.. link: 
+.. description: 
+.. type: text
+.. previewimage: /images/2021/10/re-engineering-cicd-pipelines-for-scipyre-engineering-ci-cd-scipy.png
+-->
+
+In this blog post I talk about the projects and my work during my internship at Quansight Labs. My efforts were geared towards re-engineering CI/CD pipelines for SciPy to make them more efficient to use with GitHub Actions. I also talk about the milestones that I achieved, along with the associated learnings and improvements that I made.
+
+This blog post would assume a basic understanding of CI/CD and GitHub Actions. I will also assume a basic understanding of Python and the SciPy ecosystem.
+
+<p align="center">
+    <img
+     alt="The picture displays a logo of Quansight Labs on the left and a logo of SciPy on the right. It signifies the primary purpose of the project at Quansight Labs to re-engineer the GitHub Actions CI for the SciPy and lay down the further scope for developing an entire CI matrix for build, test and release."
+     src="/images/2021/10/re-engineering-cicd-pipelines-for-scipy/re-engineering-ci-cd-scipy.png">
+    <i>Re-Engineering CI/CD pipelines for SciPy</i>
+</p>
+
+<!-- TEASER_END -->
+
+[SciPy](https://github.com/scipy/scipy) is quite an old and mature project, which is being used across the PyData community. The project currently uses a variety of continuous integration services and the Setuptools build system. As the project is growing and `distutils` will be deprecated as part of [PEP 632](https://www.python.org/dev/peps/pep-0632/), it was reasonable to migrate to a new build system. A [previous blog post](https://labs.quansight.org/blog/2021/07/moving-scipy-to-meson/) describes the migration process in detail and sets the future course for SciPy developers to look forward to faster builds.
+
+SciPy has been previously using [TravisCI](https://travis-ci.org/), [CircleCI](https://circleci.com/), and [AppVeyor](https://www.appveyor.com/) across its continuous integration pipelines. Recently, [GitHub Actions](https://github.com/features/actions), the native CI/CD service offered by GitHub, has been picked up by many projects. My project was centered around building and re-engineering the continuous integration pipelines for SciPy using GitHub Actions to be made compatible with the new build system.
```

Comment:
It would be good to point out that SciPy was already using GHA in its main repo - here we're just increasing the use of it. In https://github.com/MacPython/scipy-wheels we're not using GHA at all yet, so there's more room to improve.

Reply:
Refactored the paragraph and added a note that it was about increasing the adoption of the GHA usage.